src/graph/src/graph.ts

```diff
@@ -370,8 +378,13 @@ export class Graph implements GraphLike {
     toNode.registry = spec.registry
     toNode.dev = flags.dev
     toNode.optional = flags.optional
-    toNode.modifier = extra
-    this.addEdge(depType, spec, fromNode, toNode)
+
+    // Set the modifier if provided via extra parameter
+    if (extra) {
+      toNode.modifier = extra
+    }
```

Comment:
Not sure if this addition is relevant to this change. iirc @ruyadorno opted to always apply the modifier to the node even if it was `undefined`